hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSOutputStream.java

```diff
@@ -536,8 +536,13 @@ protected void adjustChunkBoundary() {
     }
 
     if (!getStreamer().getAppendChunk()) {
-      final int psize = (int) Math
-          .min(blockSize - getStreamer().getBytesCurBlock(), writePacketSize);
+      int psize = 0;
+      if (blockSize == getStreamer().getBytesCurBlock()) {
```

Comment:
Not think carefully, but it seems good improvement for my first feeling. cc @zhangshuyan0 Would you mind to take another review?
@hfutatzhanghb it is better to add unit tests to cover this case. Thanks.

Reply:
@Hexiaoqiao Sir, thanks for your replying. I will add unit tests soonly when i am avaliable.